src/main/java/gregtech/common/covers/CoverFluidRegulator.java

```diff
@@ -0,0 +1,248 @@
+package gregtech.common.covers;
+
+import gregtech.api.cover.ICoverable;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.*;
+import gregtech.api.util.GTFluidUtils;
+import gregtech.api.util.TextFormattingUtil;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.util.text.event.HoverEvent;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+
+public class CoverFluidRegulator extends CoverPump {
+
+    protected TransferMode transferMode;
+    protected int keepAmount = 0;
+    protected int supplyAmount = 0;
+    private static final String supplyKey = "SupplyAmount";
+    private static final String keepKey = "KeepAmount";
+
+    public CoverFluidRegulator(ICoverable coverHolder, EnumFacing attachedSide, int tier, int mbPerTick) {
+        super(coverHolder, attachedSide, tier, mbPerTick);
+        this.transferMode = TransferMode.TRANSFER_ANY;
+    }
+
+/*    @Override
+    protected int doTransferFluids(int transferLimit) {
```

Comment:
If this commented out section was handled in a different method, this commented out section could be deleted